heka/sandbox/encoders/es_fields.lua

```diff
@@ -156,6 +160,15 @@ function process_message()
         end
     end
 
+    local ok, info = pcall(cjson.decode, read_message("Fields[environment.settings]"))
+    if ok then
+        for i, k in ipairs(environment_fields) do
```

Comment:
The loop seems a little over kill but I assume you expect to be adding more values